submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoPersistenceQueryService.kt

```diff
@@ -74,7 +74,14 @@ internal class SubmissionMongoPersistenceQueryService(
     override fun getReferencedFiles(accNo: String, fileListName: String): List<ExtFile> {
         return fileListDocFileRepository
             .findAllBySubmissionAccNoAndSubmissionVersionGreaterThanAndFileListName(accNo, 0, fileListName)
-            .map { it.file.toExtFile(submissionRepo.getRelPath(accNo)) }
+            .map { it.file.toExtFile(submissionRepo.getRelPath(accNo).relPath) }
+    }
+
+    override fun findReferencedFile(accNo: String, version: Int, path: String): ExtFile? {
+        return fileListDocFileRepository
+            .findBySubmissionAccNoAndSubmissionVersionAndFilePath(accNo, version, path)
+            ?.file
+            ?.toExtFile(submissionRepo.getRelPath(accNo).relPath)
```

Comment:
I don't think so, this is calculated only once since it's a single file operation. Unless what you're suggesting is that the method should take the subRelPath as a parameter which would be kind of weird because the whole point of having the files decoupled is that we wouldn't have the submission concern at this point.

It's also my understanding that the need for a subRelPath is temporary in order to calculate the fire relPath which will be gone once the files migration is done.